src/Chart/createChart.jsx

```diff
@@ -0,0 +1,127 @@
+import React, {Component, PropTypes} from 'react'
+import 'chart.js'
+
+/* global Chart */
+
+export default function createChart (ChartElement) {
+  class ChartComponent extends Component {
+    static propTypes = {
+      data: PropTypes.object.isRequired,
+      drilldown: PropTypes.object,
+      options: PropTypes.object,
+      title: PropTypes.string
+    }
+
+    static defaultProps = {
+      drilldown: {},
+      options: {},
+      title: null
+    }
+
+    baseConfig = {
+      legend: {
+        position: 'bottom'
+      },
+      title: {
+        display: true,
+        fontSize: 18,
+        text: 'Default Area Chart'
+      }
+    }
+
+    parseObjectsFromData (data, scales) {
```

Comment:
I would probably reverse the large if/else and then bring return twice to avoid the extra indentation. I would also say that the manipulation of the data variable makes this a little more confusing than it probably really is. I don't know what datastructure that we are trying to get here by reading the code. 

```javascript
parseObjectsFromData (data, scales) {
  const {
    data: dataset,
    datasets,
    xAxis,
    yAxis
  } = data
      
  if (datasets && datasets.length > 0) {
    data.datasets = data.datasets.map((dataset) => ({
      ...dataset
    }))
    
    return data
  }
      
  const isArrayOfArrays = dataset.length > 0 && Array.isArray(dataset[0])
      
  data.datasets = []
  data.labels = data.labels || []
     
  // Go through both the x and y axis properties and
  // translate it into a dataset.
  for (let [index, dataset] of xAxis.entries()) {
    dataset = {
      ...dataset,
      data: [],
      xAxisID: dataset.xAxisID || dataset.axisId
    }
    
    if (dataset.id != null) {
      data.datasets.push(dataset)
    }
        
    // Add in the labels array to conform to the
    // chartjs framework
    if (index === 0) { // <-- Do we need to this inside of the for loop or could it be moved out since its only happening one time?
      for (const dataObject of dataset) {
        data.labels.push(dataObject[dataset.dataProperty])
      }
    }
  }
      
  for (let dataset of yAxis) {
    dataset = {
      ...dataset,
      data: [],
      yAxisID: dataset.yAxisID || dataset.axisId
    }
        
    data.datasets.push(dataset)
  }
      
  for (const dataObject of dataset) {
    for (let [index, dataset] of data.datasets.entries()) {
      // If data is in array format, add to corresponding
      // dataset in order of x-axes first then y-axes.
      if (isArrayOfArrays) {
        dataset.data.push(dataObject[index])
        
        continue
      }
          
      dataset.data.push(dataObject[dataset.dataProperty])
    }
  }
      
  return data
}
```